### PR TITLE
Update for latest version of gimli.units

### LINKED ISSUE #168: Error from gimli when importing pymt
When using the latest version of *gimli* (v0.3.2), *pymt* throws an error on import:
```python
>>> import pymt
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/mpiper/projects/pymt/pymt/__init__.py", line 1, in <module>
    from gimli import UnitSystem
ImportError: cannot import name 'UnitSystem' from 'gimli' (/Users/mpiper/anaconda3/envs/pdev/lib/python3.12/site-packages/gimli/__init__.py)
```

The Python I'm using:
```
Python 3.12.6 | packaged by conda-forge | (main, Sep 30 2024, 17:55:20) [Clang 17.0.6 ] on darwin
```

### PR DESCRIPTION
**Pull request by mdpiper**
The latest version of *gimli.units* (v0.3.2) includes some significant changes that are breaking *pymt*. This PR is an attempt to update *pymt* to work with *gimli.units*. I'm trying to make as few changes as possible to 1) get the tests passing again and 2) ensure that *gimli.units* is being used correctly in *pymt*.

This fixes #168.

### CODE REVIEW

**Comment by mdpiper** [comment]
A lot of files were updated, but most of the changes were updates from *black* and *flake8* to get get the tests passing.

I also modified the CI workflows to get them to the point of running the unit tests.

**Comment by mdpiper** [comment]
@mcflugen Would you mind taking a look at this, please? The last few tests that are failing are from changes in *gimli.units*, and I'm not sure how best to proceed.
```
=========================== short test summary info ============================
FAILED tests/framework/test_bmi_var_units.py::test_incompatible_units - gimli._udunits2.UdunitsError: udunits error (6)
FAILED tests/framework/test_bmi_var_units.py::test_bad_units - gimli._udunits2.UdunitsError: udunits error (11)
FAILED tests/test_units.py::test_get_xml - AttributeError: type object 'UnitSystem' has no attribute 'get_xml_path'
FAILED tests/test_units.py::test_default_system - AttributeError: 'str' object has no attribute 'is_file'
FAILED tests/test_units.py::test_user_system - AttributeError: 'str' object has no attribute 'is_file'
FAILED tests/test_units.py::test_env_system - AttributeError: 'str' object has no attribute 'is_file'
FAILED tests/test_units.py::test_unit_converter_bad_from_units[not_a_unit-m] - gimli._udunits2.UdunitsError: udunits error (11)
FAILED tests/test_units.py::test_unit_converter_bad_from_units[m-not_a_unit] - gimli._udunits2.UdunitsError: udunits error (11)
FAILED tests/test_units.py::test_unit_converter_bad_from_units[not_a_unit-not_a_unit] - gimli._udunits2.UdunitsError: udunits error (11)
FAILED tests/test_units.py::test_unit_converter_incompatible_units - gimli._udunits2.UdunitsError: udunits error (6)
=========== 10 failed, 352 passed, 29 skipped, 160 warnings in 5.67s ===========
```

If you could fix these, or let me know how to fix them, I'd appreciate it.

**Comment by mcflugen** [comment]
@mdpiper I've fixed the failing *gimli.units* tests. Most of those tests were simply testing *gimli*, not *pymt*, and are, in fact, repeated almost verbatim in the unit tests of the *gimli* package. As such, I just removed them as they are redundant and don't belong in *pymt*.

**NOTE: I've rebased this branch with *master* so you'll have to update your local branch before making any additional changes.**

**Comment by mdpiper** [comment]
The notebook tests are failing because _pymt_gipl_ and _pymt_ecsimplesnow_ aren't built on osx_arm64 (#172). I've updated the _ecsimplesnow_ feedstock and I'm waiting on _pymt_ecsimplesnow_. _gipl_ isn't compiling, but I think I can fix it. I'll then go through the process of updating its feedstocks.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -41,24 +41,20 @@ jobs:
...
 
-      - name: Install Unix requirements
-        if: matrix.os == 'ubuntu-latest' || matrix.os == 'macos-latest'
-        run: |
-          mamba install esmpy
-          mamba install pymt_cem pymt_child pymt_hydrotrend pymt_permamodel
+      # - name: Install Unix requirements
+      #   if: matrix.os == 'ubuntu-latest' || matrix.os == 'macos-latest'
+      #   run: |
+      #     mamba install pymt_cem pymt_child pymt_hydrotrend pymt_permamodel
```
Comment (mdpiper) [inline .github/workflows/test.yml:52]: These installs update shapely to >2, which fails because of #164.

Comment (mcflugen) [inline .github/workflows/test.yml:52]: What's the reason for not installing these components? I assume they're not working for some reason?

Comment (mdpiper) [reply .github/workflows/test.yml:52]: #164 is the problem. I revert this in e68392b094e84ab9b8a74638946ef05e8e774976, in a branch, https://github.com/mdpiper/pymt/tree/mdpiper/fixes-for-shapely-and-scipy-updates, that I was planning to submit as a PR after this one.

**Review by mcflugen — APPROVED** [review]
👍

**Pull request merged**